modules/editor/message/MessageEditor.tsx

```diff
@@ -40,35 +83,34 @@ export function MessageEditor(props: MessageEditorProps) {
 
   return useObserver(() => (
     <Stack gap={16}>
-      <div>
-        <PrimaryContentEditor message={message} form={form} />
-        <ErrorWrapper>
-          <InputError
-            error={
-              message.embedLength > 6000
-                ? "Embeds exceed 6000 character limit"
-                : undefined
-            }
-          />
-        </ErrorWrapper>
-      </div>
-      {message.embeds.map((embed, index) => (
-        <EmbedEditor
-          key={embed.id}
-          embed={embed}
-          form={form.repeatingForm("embeds").index(index)}
-        />
-      ))}
-      <div>
+      <PrimaryContentEditor message={message} form={form} />
+      <EmbedGroup message={message} form={form} />
+      <InputField
+        id={`_${message.id}_url`}
+        label="Message Link"
+        placeholder="https://discord.com/channels/..."
+        error={form.field("reference").error}
+        {...form.field("reference").inputProps}
+      >
         <PrimaryButton
-          disabled={message.size >= 10}
           onClick={() => {
-            form.repeatingForm("embeds").push({} as EmbedLike, ["timestamp"])
+            applyPatch(form.state.value, [
+              {
+                op: "remove",
+                path: form.path,
+              },
+            ])
           }}
         >
-          Add Embed
+          Remove
         </PrimaryButton>
```

Comment:
I don't think this button should be next to the message reference input, as that would lead the user to believe that it will delete the message inside of Discord itself.

Maybe a secondary button to the left of the data editor button would work, also try to hide the button for the first message in the editor.